### DISCUSSION TITLE
How do I detect the Python version at runtime? [duplicate]

### DISCUSSION THREAD

**Discussion by SuperJUNOed** [community/community#50665] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a Python file which might have to support Python versions < 3.x and >= 3.x. Is there a way to introspect the Python runtime to know the version which it is running (for example, 2.6 or 3.2.x)?

**Comment by yirideveloper** [comment, marked as answer]
Sure, take a look at sys.version and sys.version_info.

For example, to check that you are running Python 3.x, use

import sys
if sys.version_info[0] < 3:
    raise Exception("Must be using Python 3")
Here, sys.version_info[0] is the major version number. sys.version_info[1] would give you the minor version number.

In Python 2.7 and later, the components of sys.version_info can also be accessed by name, so the major version number is sys.version_info.major.

**Comment by AISoltani** [comment]
The easiest ways to do that is using this command:

`python --version`

Also for more detail and using on script you can using this :

`import sys
if not sys.version_info.major == 3 and sys.version_info.minor >= 6:

    print("Python 3.6 or higher is required.")

    print("You are using Python {}.{}.".format(sys.version_info.major, sys.version_info.minor))

    sys.exit(1)

or : 

`import platform
print(platform.python_version())`
